src/main/java/net/sf/jabref/gui/errorconsole/ErrorConsoleController.java

```diff
@@ -59,43 +58,35 @@ private void closeErrorDialog() {
      */
     private void listViewStyle() {
         // Handler for listCell appearance (example for exception Cell)
-        allMessages.setCellFactory(
-                new Callback<ListView<LogEvent>, ListCell<LogEvent>>() {
-                    @Override
-                    public ListCell<LogEvent> call(
-                            ListView<LogEvent> listView) {
-                        return new ListCell<LogEvent>() {
-
-                            @Override
-                            public void updateItem(LogEvent logMessage, boolean empty) {
-                                super.updateItem(logMessage, empty);
-                                if (logMessage != null) {
-                                    setText(logMessage.getMessage().toString());
-
-                                    Level logLevel = logMessage.getLevel();
-                                    switch (logLevel.getStandardLevel()) {
-                                        case ERROR:
-                                            getStyleClass().add("exception");
-                                            setGraphic(IconTheme.JabRefIcon.INTEGRITY_FAIL.getGraphicNode());
-                                            break;
-                                        case WARN:
-                                            getStyleClass().add("output");
-                                            setGraphic(IconTheme.JabRefIcon.INTEGRITY_WARN.getGraphicNode());
-                                            break;
-                                        case INFO:
-                                            getStyleClass().add("log");
-                                            setGraphic(IconTheme.JabRefIcon.INTEGRITY_INFO.getGraphicNode());
-                                            break;
-                                        default:
-                                            setText(null);
-                                            setGraphic(null);
-                                            break;
-                                    }
-                                }
-                            }
-                        };
+        allMessages.setCellFactory(new ViewModelListCellFactory<LogEvent>().
+                withGraphic( viewModel -> {
```

Comment:
If you feel like it, you can actually create a real ViewModel around `LogEvent`...